ludwig/modules/loss_implementations/corn.py

```diff
@@ -0,0 +1,150 @@
+# Source: https://github.com/Raschka-research-group/coral-pytorch/blob/main/coral_pytorch/losses.py
+# Sebastian Raschka 2020-2021
+# coral_pytorch
+# Author: Sebastian Raschka <sebastianraschka.com>
+#
+# License: MIT
+
+import torch
+import torch.nn.functional as F
+
+
+def coral_loss(logits, levels, importance_weights=None, reduction="mean"):
+    """Computes the CORAL loss described in.
+
+    Cao, Mirjalili, and Raschka (2020)
+    *Rank Consistent Ordinal Regression for Neural Networks
+       with Application to Age Estimation*
+    Pattern Recognition Letters, https://doi.org/10.1016/j.patrec.2020.11.008
+
+    Parameters
+    ----------
+    logits : torch.tensor, shape(num_examples, num_classes-1)
+        Outputs of the CORAL layer.
+
+    levels : torch.tensor, shape(num_examples, num_classes-1)
```

Comment:
Is it also worth adding a CoralLoss Config?

Reply:
Maybe, removed for now as not sure where it would make sense to use it.